torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -381,6 +381,48 @@ std::future<bool> launchAsyncGilCheck() {
   return resultFuture;
 }
 
+class CUDAEventCache {
```

Comment:
by convention, Can we move the class definition to .hpp file so that we can access the class anywhere in the cpp file ?

Reply:
ProcessGroupNCCL.hpp is included in 5 compilation units, whereas code in ProcessGroupNCCL.cpp only is in one, so in cases where the code is only used locally I tend to avoid the header files to keep compile times down.